### PR TITLE
Wip mgr stats

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
@liewegas @gregsfortytwo

File: src/mgr/DaemonServer.cc [hunk 1]
```diff
@@ -724,3 +729,23 @@ bool DaemonServer::handle_command(MCommand *m)
     return true;
   }
 }
+
+void DaemonServer::send_report(bool)
+{
+  auto m = new MMonMgrReport();
+  cluster_state.with_pgmap([&](const PGMap& pg_map) {
+      // FIXME: no easy way to get mon features here.  this will do for
+      // now, though, as long as we don't make a backward-incompat change.
```
Comment (gregsfortytwo) [inline src/mgr/DaemonServer.cc:738]: The manager has a monitor map; does that not expose them?

File: src/mgr/DaemonServer.cc [hunk 2]
```diff
@@ -724,3 +729,23 @@ bool DaemonServer::handle_command(MCommand *m)
     return true;
   }
 }
+
+void DaemonServer::send_report(bool)
```
Comment (gregsfortytwo) [inline src/mgr/DaemonServer.cc]: Let's not create functions with parameters we never look at?

File: src/mon/OSDMonitor.cc [hunk 3]
```diff
@@ -1076,7 +1076,10 @@ void OSDMonitor::prime_pg_temp(
   const OSDMap& next,
   pg_t pgid)
 {
-  if (mon->monmap->get_required_features().contains_all(
+  if (mon->monmap->get_required_features().empty()) {
+    dout(10) << __func__ << ": quorum not formed" << dendl;
+    return;
+  } else if (mon->monmap->get_required_features().contains_all(
```
Comment (gregsfortytwo) [inline src/mon/OSDMonitor.cc:1082]: How are we even getting into these functions without a quorum? I don't think we should have to run those checks here (although we may need to handle stale results; I'm looking at that now).

**Comment by liewegas** [comment]
it doesn't tell you encoding features; it only has methods for the 
ceph::mon::feature_t's (or whatever).  we'd need to create something along 
the lines of OSDMap::get_features() that figures out what ceph 
features are required.

File: src/mgr/DaemonServer.cc (hunk 2 above)
Comment (tchaikov) [reply src/mgr/DaemonServer.cc]: okay! we always assigned true to it before this change, so thought it was a placeholder.

File: src/mon/OSDMonitor.cc (hunk 3 above)
Comment (tchaikov) [reply src/mon/OSDMonitor.cc:1082]: @gregsfortytwo i ran into a crash in `OSDMonitor::get_trim_to()` while testing `TEST_mon_features()`, so i added all these guards.

Comment (tchaikov) [reply src/mon/OSDMonitor.cc:1082]: maybe we don't need all of them. but i wanted to test it using rados qa suites sooner, so didn't look at it closer. but will do tomorrow.

File: src/mgr/DaemonServer.cc (hunk 2 above)
Comment (tchaikov) [reply src/mgr/DaemonServer.cc]: @gregsfortytwo fixed and repushed.

**Pull request merged**